openpype/modules/sync_server/launch_hooks/pre_copy_last_published_workfile.py

```diff
@@ -178,8 +183,38 @@ def execute(self):
         )
 
         extension = last_published_workfile_path.split(".")[-1]
+
+        # Get last published workfile version
+        last_published_workfile_version = workfile_representation["context"][
+            "version"
+        ]
+
+        # Get last local workfile version
+        last_local_workfile_version = get_last_workfile_with_version(
+            get_workdir(
+                project_doc,
+                asset_doc,
+                task_name,
+                host_name,
+                anatomy=anatomy,
+                template_key=template_key,
+                project_settings=project_settings,
+            ),
+            anatomy.templates[template_key]["file"],
+            workfile_data,
+            [extension],
+        )[1]
+
+        # Use last published workfile version + 1
+        # or last local workfile version + 1 if it's higher
         workfile_data["version"] = (
-                workfile_representation["context"]["version"] + 1)
+            last_local_workfile_version if (
+                last_local_workfile_version
+                and last_local_workfile_version
+                > last_published_workfile_version
+            )
+            else last_published_workfile_version
+        ) + 1
```

Comment:
This is quite an unreadable statement.

A bit verbose but maybe this is more readable?
```python
versions = []
if last_local_workfile_version is not None:
    versions.append(last_local_workfile_version)
if last_published_workfile_version is not None:
    versions.append(last_published_workfile_version)
new_version = max(versions, default=0) + 1
```
Or:
```python
new_version = max((
    version for version in [
        last_local_workfile_version,
        last_published_workfile_version
    ]
    if version is not None
), default=0) + 1
```
But admittedly that looks like crazy mumbo jumbo again.

Reply:
```
new_version = max([
    last_local_workfile_version or 0,
    last_published_workfile_version or 0
]) + 1
```

Reply:
I went with @iLLiCiTiT implementation, but with a generator. I don't think it's worth using a list in this case.

Reply:
FYI - looking at the code I don't think that's a generator but it produces a `tuple`.